File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Phasors/Ieee1344/DigitalValue.vb
'*******************************************************************************************************
'  DigitalValue.vb - IEEE 1344 Digital value
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  11/12/2004 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Namespace Ieee1344

    <CLSCompliant(False)> _
    Public Class DigitalValue

        Inherits DigitalValueBase

        Public Sub New(ByVal parent As IDataCell, ByVal DigitalDefinition As IDigitalDefinition, ByVal value As Int16)

            MyBase.New(parent, DigitalDefinition, value)

        End Sub

        Public Sub New(ByVal parent As IDataCell, ByVal DigitalDefinition As IDigitalDefinition, ByVal binaryImage As Byte(), ByVal startIndex As Int32)

            MyBase.New(parent, DigitalDefinition, binaryImage, startIndex)

        End Sub

        Public Sub New(ByVal DigitalValue As IDigitalValue)

            MyBase.New(DigitalValue)

        End Sub

        Friend Shared Function CreateNewDigitalValue(ByVal parent As IDataCell, ByVal definition As IDigitalDefinition, ByVal binaryImage As Byte(), ByVal startIndex As Int32) As IDigitalValue

            Return New DigitalValue(parent, definition, binaryImage, startIndex)

        End Function

        Public Overrides ReadOnly Property InheritedType() As System.Type
            Get
                Return Me.GetType
            End Get
        End Property

    End Class

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IeeeC37_118/DigitalValue.cs
using System.Diagnostics;
using System;
//using PCS.Common;
using System.Collections;
using PCS.Interop;
using Microsoft.VisualBasic;
using PCS;
using System.Collections.Generic;
//using PCS.Interop.Bit;
using System.Linq;
using System.Runtime.Serialization;

//*******************************************************************************************************
//  DigitalValue.vb - IEEE C37.118 Digital value
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: VB.NET, Visual Studio 2008
//  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
//      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
//       Phone: 423/751-2827
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  11/12/2004 - J. Ritchie Carroll
//       Initial version of source generated
//
//*******************************************************************************************************


namespace PCS.PhasorProtocols
{
    namespace IeeeC37_118
    {

        [CLSCompliant(false), Serializable()]
        public class DigitalValue : DigitalValueBase
        {



            protected DigitalValue()
            {
            }

            protected DigitalValue(SerializationInfo info, StreamingContext context)
                : base(info, context)
            {


            }

            public DigitalValue(IDataCell parent, IDigitalDefinition digitalDefinition, short value)
                : base(parent, digitalDefinition, value)
            {


            }

            public DigitalValue(IDataCell parent, IDigitalDefinition digitalDefinition, byte[] binaryImage, int startIndex)
                : base(parent, digitalDefinition, binaryImage, startIndex)
            {


            }

            public DigitalValue(IDataCell parent, IDigitalDefinition digitalDefinition, IDigitalValue digitalValue)
                : base(parent, digitalDefinition, digitalValue)
            {


            }

            internal static IDigitalValue CreateNewDigitalValue(IDataCell parent, IDigitalDefinition definition, byte[] binaryImage, int startIndex)
            {

                return new DigitalValue(parent, definition, binaryImage, startIndex);

            }

            public override System.Type DerivedType
            {
                get
                {
                    return this.GetType();
                }
            }

            public new DataCell Parent
            {
                get
                {
                    return (DataCell)base.Parent;
                }
            }

            public new DigitalDefinition Definition
            {
                get
                {
                    return (DigitalDefinition)base.Definition;
                }
                set
                {
                    base.Definition = value;
                }
            }

        }

    }
}
